examples/allocation-endpoint/README.md

```diff
@@ -0,0 +1,99 @@
+```
+
+After installing Agones, deploy [ESP](https://cloud.google.com/endpoints/docs/grpc/specify-esp-v2-startup-options) which is an envoy based proxy in front of agones `agones-alloator` container. Run the following to patch the service deployement, change the service port to ESP and add annotation to `agones-allocator` service account to impersonate GCP service account. 
+
+Replace [GKE-PROJECT-ID] in `patch-agones-allocator.yaml` with your project ID before runing the scripts.
+
+```
+kubectl patch deployment agones-allocator -n agones-system --patch-file patch-agones-allocator.yaml
+kubectl patch svc agones-allocator -n agones-system --type merge -p '{"spec":{"ports": [{"port": 443,"name":"https","targetPort":9443}]}}'
+kubectl annotate sa -n agones-system agones-allocator iam.gke.io/gcp-service-account=ae-esp-sa@[PROJECT-ID].iam.gserviceaccount.com
+```
+
+## Terraform 
+The terraform modules create resources in GCP:
+
+```
+terraform apply -var "project_id=[PROJECT-ID]" -var "ae_proxy_image=[GCR-IMAGE]:[VERSION]" -var "authorized_members=[\"serviceAccount:[SERVICE-ACCOUNT-EMAIL]\"]" -var "clusters_info=[CLUSTERS-INFO]" -var "workload-pool=[GKE-PROJECT-ID].svc.id.goog"
+```
+
+`[CLUSTERS-INFO]` is in the form of `[{\"name\":\"cluster1\",\"endpoint\":\"34.83.14.82\",\"namespace\":\"default\",\"allocation_weight\":100},{...}]` deserializing to []ClusterInfo, defined in the `server/clusterselector.go`.
+
+- The `name` is a unique randomly selected name for the cluster.
+- The `endpoint` is the `agones-allocator` external IP.
+- The `namespace` is the game server namespace.
+- The `allocation_weight` is a value between 0 and 100, which sets the relative allocation rate a cluster receives compared to other clusters. By setting weight to zero, a cluster stops receiving allocation requests.
+
+`[GCR-IMAGE]` has a default set to `gcr.io/agones-images/allocation-endpoint-proxy:latest`.
+
+`[SERVICE-ACCOUNT-EMAIL]` is the service account to be granted access the Allocation Endpoint.
+
+## Server
+
+The Allocation Endpoint proxy code is in `./server` folder. You can make changes and run the following to build and push the image to your own GCR repository:
+
+```
+docker build --tag gcr.io/[PROJECT-ID]/allocation-endpoint-proxy:[VERSION] .
+docker push gcr.io/[PROJECT-ID]/allocation-endpoint-proxy:[VERSION]
+```
+
+## Client
+
+The Allocation Endpoint client code is in `./client` folder. Get the Service Account Key for one of the Service Accounts in the list of `authorized_members` and put it under `sa_key.json`. Alternatively, you can leverage [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to retrieve the access token using default service account from metadata serer, when deploying your client in a GCP solution, e.g. GKE.
+
+```
+go run *.go --url=[CLOUD-RUN-ENDPOINT]
+
+```
+
+`[CLOUD-RUN-ENDPOINT]` is the cloud run endpoint printed out after running the terraform.
```

Comment:
Terraform output this for me:

```
Outputs:

endpoint = https://allocation-endpoint-proxy-stnq4mbela-uc.a.run.app
```

When I tried to run `go run *.go --url=https://allocation-endpoint-proxy-stnq4mbela-uc.a.run.app` I got the error `"transport: Error while dialing dial tcp: address https://allocation-endpoint-proxy-stnq4mbela-uc.a.run.app:443: too many colons in address"`. 

I had to drop the `https://` prefix and it worked:

`go run *.go --url=allocation-endpoint-proxy-stnq4mbela-uc.a.run.app`

Reply:
It looks like the code prepends the `https://` on line 64 but this isn't clear from the comment here or the flag description ("The Cloud Run URL for allocation endpoint."). When I see "URL" I think it means to include the scheme but the code assumes that it isn't included.

Reply:
Thanks! I added a note.